### PR TITLE
ML-027/add person system

### PR DESCRIPTION
**Pull request by KarinaKlevzhits**
https://trello.com/c/94e5q0WU/40-ml-027-add-person-system

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -30,3 +30,8 @@
             username: Faker::Internet.unique.username(specifier: 8), birthday: Faker::Date.birthday(min_age: 18, max_age: 65),
             role_id: reviewer_role.id, confirmed_at: Time.zone.now)
 FactoryBot.create_list(:rating, 20)
+Role.create(name: 'admin', permission_attributes: { can_lock_user: true, can_edit_role: true, can_set_role: true, can_read: true, can_edit: true })
+Role.create(name: 'redactor', permission_attributes: { can_lock_user: false, can_edit_role: false, can_set_role: false, can_read: true, can_edit: true })
+Role.create(name: 'reviewer', permission_attributes: { can_lock_user: false, can_edit_role: false, can_set_role: false, can_read: true, can_edit: false })
```
Comment (WhatWasThatt) [inline db/seeds.rb:35]: these roles are already created above

File: spec/factories/staff_types.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+# Indexes
+#
+#  index_staff_types_on_name  (name) UNIQUE
+#
+FactoryBot.define do
+  factory :staff_type do
+    name { Faker::Emotion.unique.noun }
```
Comment (WhatWasThatt) [inline spec/factories/staff_types.rb:16]: I think it makes more sense to use Faker::Company.unique.profession

File: app/controllers/api/v1/staff_types_controller.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+      def index
+        @staff_types = Dictionary::FindService.new(StaffType, params).call
+
+        render json: @staff_types, each_serializer: DictionarySerializer
+      end
+
+      def show
+        render json: @staff_type, serializer: DictionarySerializer
+      end
```
Comment (WhatWasThatt) [inline app/controllers/api/v1/staff_types_controller.rb:14]: I think we can abandon the show method

**Review by WhatWasThatt — APPROVED** [review]
Great job!

File: app/models/movie.rb [hunk 4]
```diff
@@ -39,4 +43,34 @@ class Movie < ApplicationRecord
...
+
+  def screencasters
+    staffs_by_occupation(6)
+  end
+
+  private
+
+  def staffs_by_occupation(occupation)
+    staffs.where(movie_staffs: { staff_type_id: occupation })
+  end
```
Comment (dmitrymalllpr) [inline app/models/movie.rb:75]: rework please. Write me please if you don t have ideas

**Review by dmitrymalllpr — CHANGES_REQUESTED** [review]

File: app/models/movie.rb (hunk 4 above)
Comment (KarinaKlevzhits) [reply app/models/movie.rb:75]: can you concrete your comment?)

Comment (dmitrymalllpr) [reply app/models/movie.rb:75]: okay
1) for me staff_type looks like enum
2) what about scopes?
3) avoid magic numbers

**Review by dmitrymalllpr — APPROVED** [review]
Good job and Good night!

**Review by twill11 — APPROVED** [review]
GJ!

**Pull request merged**